superset-frontend/src/views/CRUD/welcome/ActivityTable.tsx

```diff
@@ -79,31 +97,68 @@ const ActivityContainer = styled.div`
   }
 `;
 
+const UNTITLED = t('[Untitled]');
+const UNKNOWN_TIME = t('Unknown');
+// translation keys for last action on
+const TRANS_LAST_VIEWED = `Last viewed %s`;
+const TRANS_LAST_MODIFIED = `Last modified %s`;
+
+const getEntityTitle = (e: ActivityObject) => {
+  if ('dashboard_title' in e) return e.dashboard_title || UNTITLED;
+  if ('slice_name' in e) return e.slice_name || UNTITLED;
+  if ('label' in e) return e.label || UNTITLED;
```

Comment:
This fixes the case when saved queries don't have a title (label).

### Before

![Snip20210222_37](https://user-images.githubusercontent.com/335541/108794269-4a4c4f00-753a-11eb-9214-dcf94a3c172b.png)

### After

![Snip20210222_39](https://user-images.githubusercontent.com/335541/108794282-52a48a00-753a-11eb-98e6-e831ce93ecc8.png)